records/README.md

```diff
@@ -0,0 +1,220 @@
+
+### Merkle DAG and IPFS Objects
+
+A merkle dag is a directed acyclic graph whose links are (a) hashes of the edge target, and (b) contained within the edge source. (syn. merkle tree, hash tree)
+
+In this spec, _the merkle dag_ (specific one) refers to the IPFS merkle dag. _IPFS Object_ refers to objects in the merkle dag, which follow the IPFS merkledag format. (Read those specs)
+
+
+## Constraints
+
+IPRS has the following _hard_ constraints:
+
+- **MUST** be transport agnostic. (transport refers to how computers communicate).
+- **MUST** be replication agnostic. (replication refers to the protocol computers use to transfer and propagate whole records and other objects)
+- **MUST** be validity scheme agnostic. (validity scheme includes PKI, TI, and other "agreed upon" trusted infrastructure)
+- **MUST** be trustless: no trusted third parties are imposed by IPRS (though some may be adopted by a validity scheme. e.g. root CAs in the CA system PKI, or a blockchain in a blockchain TI). In most cases, users may have to trust each other (as they must trust the record value -- e.g. DNS), but in some cases there may be cryptographic schemes that enable full trustlessness.
+
+
+It is easy to be agnostic to transport, replication, and validity scheme as long as users can expect to control or agree upon the programs or protocols used in concert with IPRS. Concretely, the user can select specific transports or validity schemes to suit the user's application constraints. It is the user's responsibility to ensure both record crafters and verifiers agree upon these selections.
+
+## Construction
+
+### The Objects
+
+IPRS records are expressed as [merkledag](../merkledag) objects. This means that the records are linked authenticated data structures, and can be natively replicated over IPFS itself and other merkledag distribution systems.
+
+The objects:
+
+- A `Record` object expresses a value, a validity scheme, and validity data.
+- A `Signature` object could be used to sign and authenticate a record.
+- An `Encryption` object could be used to encrypt a record.
+
+```go
+Record Node {
+  Scheme   Link // link to a validity scheme
+  Value    Link // link to an object representing the value.
+  Validity Data // data needed to satisfy the validity scheme
+}
+```
+
+To achieve good performance, record storage and transfer should bundle all the necessary objects and transmit them together. While "the record object" is only one of the dag objects, "the full record" means a bundle of all objects needed to fully represent, verify, and use the record. (This recommendation does not necessarily include data that records _describe_, for example an ipfs provider record (which signals to consumers that certain data is available) would not include the data itself as part of "the full record").
```

Comment:
Maybe drop “and use” from the “full record” definition, since that's so wishy.  To get past verification you'll need the Record, linked Scheme, and any further objects needed to validate the Validity data (which you may also want to store as links?  E.g. to public keys, etc?).  The only thing the “full record” does not need to bundle is stuff down the Value link.